### PR TITLE
[POAE7-2346] using new data format

### PR DESCRIPTION
**Pull request by YantingTao1315**
### What changes were proposed in this pull request?
+ enable new data format by using gflags ```--use_cider_data_format```
+ provide test method but cannot ensure all test will past when using new data format.

### Why are the changes needed?
+ test demands

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
local test by setting gflags

### CODE REVIEW

File: cider/include/cider/CiderRuntimeModule.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ class CiderRuntimeModule {
   size_t getGroupByAggHashTableBufferNum() const;
   bool isGroupBy() const;
   bool hasCountDistinct() const;
+  bool isUsingArrowFromat() const;
```
Comment (jikunshang) [inline cider/include/cider/CiderRuntimeModule.h:63]: I think we should not expose such API. User side should know what compileOption/runtimeOption is.

File: cider/include/cider/CiderRuntimeModule.h [hunk 2]
```diff
@@ -150,6 +151,7 @@ class CiderRuntimeModule {
   constexpr static size_t kMaxOutputRows = 1000;
 
   bool is_group_by_;
+  bool is_using_arrow_fromat_;
```
Comment (kd97) [inline cider/include/cider/CiderRuntimeModule.h:154]: can not get from compileOption?

File: cider/include/cider/CiderRuntimeModule.h (hunk 1 above)
Comment (YantingTao1315) [reply cider/include/cider/CiderRuntimeModule.h:63]: Agree, I will set a private member in operator

File: cider/include/cider/CiderRuntimeModule.h (hunk 2 above)
Comment (YantingTao1315) [reply cider/include/cider/CiderRuntimeModule.h:154]: operator doesn't save compileOption,
compileOption is a private member of Runtimemudule,
so will add a private member in operator, and set it when initialization

File: cider-velox/src/CiderOperator.cpp [hunk 3]
```diff
@@ -90,9 +96,23 @@ void CiderOperator::addInput(RowVectorPtr input) {
   }
 
   input_ = std::move(input);
-  auto inBatch =
-      dataConvertor_->convertToCider(input_, input_->size(), &convertorInternalCounter);
-  ciderRuntimeModule_->processNextBatch(inBatch);
+  if (is_using_arrow_fromat_) {
+    for (size_t i = 0; i < input_->childrenSize(); i++) {
+      input_->childAt(i)->mutableRawNulls();
```
Comment (xieqi) [inline cider-velox/src/CiderOperator.cpp:101]: What is this loop for?

**Review by xieqi — DISMISSED** [review]

File: cider-velox/src/CiderOperator.cpp (hunk 3 above)
Comment (Ma-Jian1) [reply cider-velox/src/CiderOperator.cpp:101]: It's so weird

File: cider-velox/src/CiderStatefulOperator.cpp [hunk 4]
```diff
@@ -43,6 +45,13 @@ RowVectorPtr CiderStatefulOperator::getOutput() {
   if (ret == CiderRuntimeModule::ReturnCode::kNoMoreOutput) {
     finished_ = true;
   }
+  if (is_using_arrow_fromat_) {
+    auto [schema, array] = output_->move();
+    VectorPtr baseVec = importFromArrowAsOwner(*schema, *array, operatorCtx_->pool());
+    CiderBatchUtils::freeArrowArray(array);
+    CiderBatchUtils::freeArrowSchema(schema);
+    return std::reinterpret_pointer_cast<RowVector>(baseVec);
```
Comment (Ma-Jian1) [inline cider-velox/src/CiderStatefulOperator.cpp:53]: move() allocate new `schema`, `array`, and then `freed` immediately, not perfect.

File: cider-velox/src/CiderOperator.cpp (hunk 3 above)
Comment (YantingTao1315) [reply cider-velox/src/CiderOperator.cpp:101]: workaround, process cannot support no null buffer

File: cider-velox/src/CiderStatefulOperator.cpp (hunk 4 above)
Comment (YantingTao1315) [reply cider-velox/src/CiderStatefulOperator.cpp:53]: add a new move method

File: cider-velox/src/CiderOperator.cpp [hunk 5]
```diff
@@ -58,6 +62,8 @@ CiderOperator::CiderOperator(int32_t operatorId,
         ciderCompileResult, compile_option, exec_option, allocator);
     outputSchema_ = std::make_shared<CiderTableSchema>(
         ciderCompileResult->getOutputCiderTableSchema());
+
+    is_using_arrow_fromat_ = compile_option.use_cider_data_format;
```
Comment (jikunshang) [inline cider-velox/src/CiderOperator.cpp:66]: typo: `is_using_arrow_format_`

**Review by Ma-Jian1 — DISMISSED** [review]

**Comment by winningsix** [comment]
@YantingTao1315 can you resolve the conflicts?

**Review by bigPYJ1151 — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**